ductus/tools/utils.py

```diff
@@ -566,3 +566,53 @@ def filter_experiment(sample_sheet_file):
         return False
     else:
         return True
+    
+
+def get_no_expected_fastqs(sample_sheet_file, file_list):
+    """
+        This function uses the information in a sample sheet to calculate the expected
+        number of fastq-files that should be transferred from the instrument to
+        the hospital file area. It uses the lane information available in the sample sheet, 
+        i.e. this is only applicable for sample sheet version 2. The number of expected
+        fastq-files is compared to the length of a list of files generated with the expression
+        glob.glob("<% ctx(fastq_files_path) %>/**/*.fastq.gz", recursive=True). 
+
+        - The function should return True, i.e. continue processing without any further
+        checks, if the sample sheet is v1. 
+
+        - If the sample sheet is detected as version 2 the expected number of fastq-files 
+        will be calculated and compared to the list of files. If the expected number of
+        files is found the return value us True, 
+        otherwise false (to enable retry until all expected files are transferred).
+        
+        param sample_sheet_file: string
+        param file_list: list
+        return: boolean
+    """
+    
+    if is_old_ductus_format(sample_sheet_file):
+        return True
+    else:
+        with open(sample_sheet_file) as file:
+            line = file.readline()
+            while not line.startswith("Lane,"):
+                line = file.readline()
+            
+            nr_of_samples = 0
+            lane_count = []
+            line = file.readline()
+            while line.startswith(("1,", "2,", "3,", "4,", "5,", "6,", "7,", "8,")):
```

Comment:
Use regular expression to ensure future proof with more than 8 lanes?
import re
```suggestion
            while re.search("^[0-9]+,", line):
```

Reply:
I have noticed that our sample sheets can look a bit different when it comes to the information after the [Data]- section. Sometimes it is empty lines and sometimes it is repeated info so I wanted to make it as specific as possible to not risk to read an extra line or so. But if we can ensure that this wont happen the regexp solution is much cleaner.